File: app/src/main/java/me/chrislane/accudrop/task/FetchLastJumpIdTask.java
package me.chrislane.accudrop.task;

import android.os.AsyncTask;

import me.chrislane.accudrop.viewmodel.DatabaseViewModel;

public class FetchLastJumpIdTask extends AsyncTask<Void, Void, Integer> {

    private final DatabaseViewModel databaseViewModel;
    private final Listener listener;

    public FetchLastJumpIdTask(FetchLastJumpIdTask.Listener listener, DatabaseViewModel databaseViewModel) {
        this.listener = listener;
        this.databaseViewModel = databaseViewModel;
    }

    @Override
    protected Integer doInBackground(Void... voids) {
        return databaseViewModel.getLastJumpId();
    }

    @Override
    protected void onPostExecute(Integer jumpId) {
        super.onPostExecute(jumpId);
        listener.onFinished(jumpId);
    }

    public interface Listener {
        void onFinished(Integer jumpId);
    }
}

File: app/src/main/java/me/chrislane/accudrop/task/FetchLastJumpIdTask.kt
package me.chrislane.accudrop.task

import android.os.AsyncTask

import me.chrislane.accudrop.viewmodel.DatabaseViewModel

class FetchLastJumpIdTask(private val listener: (Int?) -> Unit, private val databaseViewModel: DatabaseViewModel) : AsyncTask<Void, Void, Int>() {

    override fun doInBackground(vararg voids: Void): Int? {
        return databaseViewModel.lastJumpId
    }

    override fun onPostExecute(jumpId: Int?) {
        super.onPostExecute(jumpId)
        listener(jumpId)
    }
}
